src/H5VLnative_file.c

```diff
@@ -740,21 +740,35 @@ H5VL__native_file_optional(void *obj, H5VL_optional_args_t *args, hid_t H5_ATTR_
  * Return:      Success:    SUCCEED
  *              Failure:    FAIL (file will not be closed)
  *
+ * Modifications:
+ *
+ *              BMR -- 9/15/21
+ *              Temporary solution for CVE-2020-10812:
+ *                  Added a check for the returned value of H5F_NREFS to detect
+ *                  a file shared pointer being null possibly due to file
+ *                  corruption.  When that situation occurs, H5F_NREFS will return
+ *                  UINT_MAX.  This is to avoid core dump in production mode.
  *-------------------------------------------------------------------------
  */
 herr_t
 H5VL__native_file_close(void *file, hid_t H5_ATTR_UNUSED dxpl_id, void H5_ATTR_UNUSED **req)
 {
-    int    nref;
-    H5F_t *f         = (H5F_t *)file;
-    hid_t  file_id   = H5I_INVALID_HID;
-    herr_t ret_value = SUCCEED; /* Return value */
+    int      nref;
+    H5F_t *  f         = (H5F_t *)file;
+    hid_t    file_id   = H5I_INVALID_HID;
+    unsigned f_nrefs   = UINT_MAX;
+    herr_t   ret_value = SUCCEED; /* Return value */
 
```

Comment:
I'm a bit confused about H5I_get_ref() and H5F_NREFS -> H5F_get_nrefs, please confirm that the use of these two functions here are correct.